packages/sanity/src/_internal/cli/actions/schema/__tests__/storeManifestSchemasAction.test.ts

```diff
@@ -0,0 +1,171 @@
+import {readFileSync} from 'node:fs'
+
+import {type CliCommandArguments, type CliCommandContext} from '@sanity/cli'
+import {beforeEach, describe, expect, it, type Mock, vi} from 'vitest'
+
+import storeManifestSchemas from '../storeSchemasAction'
+
+// Mock dependencies
+vi.mock('node:fs', () => ({
+  readFileSync: vi.fn((path) => {
+    if (path.includes('create-manifest.json')) {
+      return JSON.stringify({
+        workspaces: [
+          {
+            name: 'testWorkspace',
+            projectId: 'testProjectId',
+            dataset: 'testDataset',
+            schema: 'testSchema.json',
+          },
+        ],
+      })
+    }
+    if (path.includes('testSchema.json')) {
+      return JSON.stringify({
+        _type: 'schema',
+        name: 'testSchema',
+      })
+    }
+    throw new Error('File not found')
+  }),
+}))
+
+type SpinnerInstance = {
+  start: Mock<() => SpinnerInstance>
+  succeed: Mock<() => SpinnerInstance>
+  fail: Mock<() => SpinnerInstance>
+  info: Mock<() => SpinnerInstance>
+  text: string
+  prefixText: string
+  suffixText: string
+  color: string
+}
+
+describe('storeManifestSchemas', () => {
+  let mockContext: CliCommandContext
+  let spinnerInstance: SpinnerInstance
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+
+    spinnerInstance = {
+      start: vi.fn(() => spinnerInstance),
+      succeed: vi.fn(() => spinnerInstance),
+      fail: vi.fn(() => spinnerInstance),
+      info: vi.fn(() => spinnerInstance),
+      text: '',
+      prefixText: '',
+      suffixText: '',
+      color: '',
+    }
+
+    mockContext = {
+      output: {
+        spinner: vi.fn().mockReturnValue(spinnerInstance),
+        error: vi.fn(),
+        success: vi.fn(),
+      },
+      workDir: '/path/to/workdir',
+      apiClient: vi.fn(() => ({
+        config: () => ({projectId: 'testProjectId'}),
+        withConfig: vi.fn().mockReturnThis(),
+        transaction: vi.fn().mockReturnThis(),
+        createOrReplace: vi.fn().mockReturnThis(),
+        commit: vi.fn().mockResolvedValue({}),
+      })),
+    } as unknown as CliCommandContext
+  })
+
+  it('should store all schemas when no flags are provided', async () => {
+    const args: CliCommandArguments<any> = {
```

Comment:
I think the tests would be bit more succinct if you made a helper function to just pass the non-common argumens.

Ie `const args = getStoreCommandArgs({extOptions: {workspace: 'testWorkspace'}})`

At first I wonder if all these empty arrays where important for the test-case, but i see its duplicated throughout.